libs/product/src/reducers/composite-product-entities/composite-product-entity.ts

```diff
@@ -3,5 +3,11 @@ import { DaffCompositeProductItemOption } from '../../models/composite-product-i
 
 export interface DaffCompositeProductEntity {
 	id: string;
-	items: Dictionary<DaffCompositeProductItemOption['id']>;
+	items: DaffCompositeProductEntityItem[];
```

Comment:
I don't understand why this needs to be an array. Is order significant?

Reply:
oh, I guess it could be:
```
{
    itemIdValue: {
      value: optionId,
      qty: quantity
   }
}
```
I'll change it to that. Order doesn't matter.

Reply:
It seems that its much simpler to deal with it as a dictionary. If you need the selector to be an array I would just convert it to array in the selector.

Reply:
changed back to dictionary. I force pushed to the branch. The changes are closer to what it was before the original commit.